lang/csharp/src/apache/main/Schema/EnumSchema.cs

```diff
@@ -47,6 +48,33 @@ public class EnumSchema : NamedSchema
         /// </summary>
         public int Count { get { return Symbols.Count; } }
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="EnumSchema"/> class.
+        /// </summary>
+        /// <param name="name">name of enum.</param>
+        /// <param name="space">namespace of enum.</param>
+        /// <param name="aliases">list of aliases for the name.</param>
+        /// <param name="symbols">list of enum symbols.</param>
+        /// <param name="customProperties">custom properties on this schema.</param>
+        /// <param name="doc">documentation for this named schema.</param>
+        /// <param name="defaultSymbol"></param>
+        public EnumSchema(string name,
```

Comment:
Schema classes already use a Factory pattern method "NewInstance".  I would recommend following the same pattern than exposing the constructor.

Reply:
NewInstance is internal, and it's more of a deserializer than an object creator, it contains a lot of logic. I can change the public constructors to a factory method (how should I call it?), is that the 'interface' we want the schema classes to expose? I think constructor is a bit more convenient to use.